numpy/ma/extras.py

```diff
@@ -1440,60 +1440,7 @@ class MAxisConcatenator(AxisConcatenator):
     mr_class
 
     """
-
-    def __init__(self, axis=0):
-        AxisConcatenator.__init__(self, axis, matrix=False)
-
-    def __getitem__(self, key):
-        if isinstance(key, str):
-            raise MAError("Unavailable for masked array.")
```

Comment:
Actually,this path makes no sense for masked arrays, since it returns a plain np.matrix - I'll add back the check